x/tss/keeper/grpc_query.go

```diff
@@ -58,3 +62,72 @@ func (q Querier) AssignableKey(c context.Context, req *types.AssignableKeyReques
 
 	return &types.AssignableKeyResponse{Assignable: !assigned}, nil
 }
+
+// ValidatorMultisigKeys returns a map of active multisig role key ids to pub keys of a validator
+func (q Querier) ValidatorMultisigKeys(c context.Context, req *types.ValidatorMultisigKeysRequest) (*types.ValidatorMultisigKeysResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	valAddress, err := sdk.ValAddressFromBech32(req.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	validator := q.staker.Validator(ctx, valAddress)
+	if validator == nil {
+		return nil, fmt.Errorf("not a validator")
+	}
+
+	chains := q.nexus.GetChains(ctx)
+	keys := make(map[string]*types.ValidatorMultisigKeysResponse_Keys, 10)
+
+	keyRoles := []exported.KeyRole{exported.MasterKey, exported.SecondaryKey}
+
+	for _, chain := range chains {
+		for _, keyRole := range keyRoles {
+			if currentKey, found := q.keeper.GetCurrentKey(ctx, chain, keyRole); found {
+				if valKeys, ok := q.getSerializedMultisigKeys(ctx, currentKey, valAddress); ok {
+					keys[string(currentKey.ID)] = valKeys
+				}
+			}
+
+			if nextKey, found := q.keeper.GetNextKey(ctx, chain, keyRole); found {
+				if valKeys, ok := q.getSerializedMultisigKeys(ctx, nextKey, valAddress); ok {
+					keys[string(nextKey.ID)] = valKeys
+				}
+			}
+
+			oldActiveKeys, err := q.keeper.GetOldActiveKeys(ctx, chain, keyRole)
+			if err != nil {
+				return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, err.Error())
+			}
+
+			for _, oldActiveKey := range oldActiveKeys {
+				if valKeys, ok := q.getSerializedMultisigKeys(ctx, oldActiveKey, valAddress); ok {
+					keys[string(oldActiveKey.ID)] = valKeys
+				}
+			}
+		}
+	}
+
+	return &types.ValidatorMultisigKeysResponse{Keys: keys}, nil
```

Comment:
```suggestion
        chains := q.nexus.GetChains(ctx)
	keyRoles := []exported.KeyRole{exported.MasterKey, exported.SecondaryKey}

	var keys []exported.Key
	for _, chain := range chains {
		for _, keyRole := range keyRoles {
			if currentKey, found := q.keeper.GetCurrentKey(ctx, chain, keyRole); found {
				keys = append(keys, currentKey)
			}

			if nextKey, found := q.keeper.GetNextKey(ctx, chain, keyRole); found {
				keys = append(keys, nextKey)
			}

			oldActiveKeys, err := q.keeper.GetOldActiveKeys(ctx, chain, keyRole)
			if err != nil {
				return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, err.Error())
			}

			keys = append(keys, oldActiveKeys...)
		}
	}

	res := types.ValidatorMultisigKeysResponse{Keys: make(map[string]*types.ValidatorMultisigKeysResponse_Keys, len(keys))}
	for _, key := range keys {
		valKeys, ok := q.getSerializedMultisigKeys(ctx, key, valAddress)
		if !ok {
			continue
		}

		res.Keys[string(key.ID)] = valKeys
```
less duplicated code and also less nested blocks.